jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamDebugSupport.kt

```diff
@@ -45,7 +56,35 @@ interface SamDebugSupport {
         state: SamRunningState,
         debugHost: String,
         debugPorts: List<Int>
-    ): Promise<XDebugProcessStarter?> = resolvedPromise(createDebugProcess(environment, state, debugHost, debugPorts))
+    ): Promise<XDebugProcessStarter?> {
+        val promise = AsyncPromise<XDebugProcessStarter?>()
+        val bgContext = ExpirableExecutor.on(AppExecutorUtil.getAppExecutorService()).expireWith(environment).coroutineDispatchingContext()
+
+        ApplicationThreadPoolScope(environment.runProfile.name).launch(bgContext) {
```

Comment:
we're still reusing the application pool, but is it okay to spin up a scope like this?

Reply:
Yep, this is how dotnet does it (this line was copied out of DotnetSamDebugSupport)